admin_ui/src/utils.ts

```diff
@@ -104,34 +125,44 @@ export function convertFormValue(params: {
 
     if (value == "null") {
         value = null
-    } else if (schema?.properties[key].type == "array") {
+    } else if (schema.properties[key]["anyOf"][0].type == "array") {
```

Comment:
Ideally we could do with some kind of function for doing this match. It's unlikely to happen but if the element we need isn't at the first index it would fail.

Something like:

```javascript
const hasType(property, type: string): boolean => {
    const types = property.anyOf.map((i) => i.type)
    return types.indexOf(type) != -1
}
```

Reply:
Hmm, some of them have three potential types:

<img width="493" alt="Screenshot 2023-09-14 at 22 40 15" src="https://github.com/piccolo-orm/piccolo_admin/assets/350976/e5ff8b72-c782-4d28-906d-9ce5ddba8b11">

Reply:
If they're in order, then maybe getting the first element is the way to go.

I wonder what happens if we have a Piccolo column with `required=True` - does it still return `anyof`?

Reply:
No. If `required=true`, not `anyOf` is returned.

Reply:
> Ideally we could do with some kind of function for doing this match. It's unlikely to happen but if the element we need isn't at the first index it would fail.
> 
> Something like:
> 
> ```js
> const hasType = (property, type: string): boolean => {
>     const types = property.anyOf.map((i) => i.type)
>     return types.indexOf(type) != -1
> }
> ```

In fact, for all columns whose `type` or `format` defined in Piccolo ORM [pydantic.py](https://github.com/piccolo-orm/piccolo/blob/4f898c1a2a971feb7bd77e36bfe7fc14dd7a5108/piccolo/utils/pydantic.py#L300-L353), we don't need `anyOf`. I will make a new commit with those changes. If it's not good, feel free to change it as you think is best.